### PR TITLE
AiService support @Profile

### PR DESCRIPTION
**Pull request by Martin7-1**
Close [langchain4j#1430](https://github.com/langchain4j/langchain4j/issues/1430)

1. Support `@AiService` with `@Profile`.

### CODE REVIEW

File: langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServicesAutoConfig.java [hunk 1]
```diff
@@ -59,6 +64,15 @@ BeanFactoryPostProcessor aiServicesRegisteringBeanFactoryPostProcessor() {
             for (String aiService : aiServices) {
                 Class<?> aiServiceClass = beanFactory.getType(aiService);
 
+                // Check profile
+                if (aiServiceClass.isAnnotationPresent(Profile.class)) {
+                    Profile profileAnnotation = aiServiceClass.getAnnotation(Profile.class);
+                    String[] profiles = profileAnnotation.value();
+                    if (!environment.matchesProfiles(profiles)) {
```
Comment (langchain4j) [inline langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServicesAutoConfig.java:71]: is this filtering really required twice? (also in `AiServiceScannerProcessor`)

**Review by langchain4j — COMMENTED** [review]
@Martin7-1 thank you!

File: langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServicesAutoConfig.java (hunk 1 above)
Comment (Martin7-1) [reply langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServicesAutoConfig.java:71]: Hmm.. Good question. Maybe we only need to filter it in `AiServiceScannerProcessor` to skip scanning the BeanDefinition with AiService.

Comment (Martin7-1) [reply langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServicesAutoConfig.java]: Done. Now only the `AiServiceScannerProcessor` do filtering with `@Profile`

**Comment by Martin7-1** [comment]
@langchain4j At first I did want to do filtering in `AiServiceAutoConfig`, but when I implemented it I ran into: `Failed to instantiate [dev.langchain4j.service.spring.mode.automatic.withProfiles. AiServiceWithProfiles]: Specified class is an interface`. It seems that `Spring Boot` still wants to find an implementation of `@AiService~ and register it. so I think we should filter it during the registration phase of the `BeanDefinition.`

I haven't found a way to do this in `AiServiceAutoConfig` yet, let me know if you know one.

File: langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServiceScannerProcessor.java [hunk 2]
```diff
@@ -63,4 +67,29 @@ private Set<String> getBasePackages(ConfigurableListableBeanFactory beanFactory)
...
+                        Class<?> beanClass = Class.forName(registry.getBeanDefinition(beanName).getBeanClassName());
+                        if (beanClass.isAnnotationPresent(AiService.class) && beanClass.isAnnotationPresent(Profile.class)) {
+                            Profile profileAnnotation = beanClass.getAnnotation(Profile.class);
+                            String[] profiles = profileAnnotation.value();
+
+                            return !environment.matchesProfiles(profiles);
+                        } else {
+                            return false;
+                        }
+                    } catch (Exception e) {
```
Comment (langchain4j) [inline langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServiceScannerProcessor.java:84]:
```suggestion
                    } catch (ClassNotFoundException e) {
```

File: langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServiceScannerProcessor.java [hunk 3]
```diff
@@ -63,4 +67,29 @@ private Set<String> getBasePackages(ConfigurableListableBeanFactory beanFactory)
...
+                        // TODO
```
Comment (langchain4j) [inline langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiServiceScannerProcessor.java:85]:
```suggestion
```

**Review by langchain4j — APPROVED** [review]
@Martin7-1 great job, thank you!

**Pull request merged**